services/brig/src/Brig/ZAuth.hs

```diff
@@ -90,52 +103,68 @@ runZAuth :: MonadIO m => Env -> ZAuth a -> m a
 runZAuth e za = liftIO $ runReaderT (unZAuth za) e
 
 data Settings = Settings
-    { _keyIndex             :: !Int                   -- ^ Secret key index to use
-                                                      --   for token creation
-    , _userTokenTimeout     :: !UserTokenTimeout      -- ^ User token validity timeout
-    , _sessionTokenTimeout  :: !SessionTokenTimeout   -- ^ Session token validity timeout
-    , _accessTokenTimeout   :: !AccessTokenTimeout    -- ^ Access token validity timeout
-    , _providerTokenTimeout :: !ProviderTokenTimeout  -- ^ Access token validity timeout
+    { _keyIndex                    :: !Int                         -- ^ Secret key index to use
+                                                                   --   for token creation
+    , _userTokenTimeout            :: !UserTokenTimeout            -- ^ User token validity timeout
+    , _sessionTokenTimeout         :: !SessionTokenTimeout         -- ^ Session token validity timeout
+    , _accessTokenTimeout          :: !AccessTokenTimeout          -- ^ Access token validity timeout
+    , _providerTokenTimeout        :: !ProviderTokenTimeout        -- ^ Proider token validity timeout
+    , _legalHoldUserTokenTimeout   :: !LegalHoldUserTokenTimeout   -- ^ Legal Hold User token validity timeout
+    , _legalHoldAccessTokenTimeout :: !LegalHoldAccessTokenTimeout -- ^ Legal Hold Access token validity timeout
     } deriving (Show, Generic)
 
 defSettings :: Settings
 defSettings = Settings 1
-    (UserTokenTimeout (60 * 60 * 24 * 28))    -- 28 days
-    (SessionTokenTimeout (60 * 60 * 24))      -- 1 day
-    (AccessTokenTimeout 900)                  -- 15 minutes
-    (ProviderTokenTimeout (60 * 60 * 24 * 7)) -- 7 days
+    (UserTokenTimeout (60 * 60 * 24 * 28))          -- 28 days
+    (SessionTokenTimeout (60 * 60 * 24))            -- 1 day
+    (AccessTokenTimeout 900)                        -- 15 minutes
+    (ProviderTokenTimeout (60 * 60 * 24 * 7))       -- 7 days
+    (LegalHoldUserTokenTimeout (60 * 60 * 24 * 56)) -- 56 days
+    (LegalHoldAccessTokenTimeout (60 * 15 ))        -- 15 minutes
 
 data Env = Env
     { _private  :: !ZC.Env
     , _public   :: !ZV.Env
     , _settings :: !Settings
     }
 
-type AccessToken   = Token Access
-type UserToken     = Token User
-type ProviderToken = Token Provider
-type BotToken      = Token Bot
+type AccessToken          = Token Access
+type UserToken            = Token User
+type ProviderToken        = Token Provider
+type BotToken             = Token Bot
+type LegalHoldUserToken   = Token LegalHoldUser
+type LegalHoldAccessToken = Token LegalHoldAccess
```

Comment:
All these type aliases don't seem to be any shorter than their definitions, only more opaque.  `FUTUREWORK` to remove them and inline the defs?

Reply:
I suppose it saves parentheses. And reads more naturally. Not sure I agree to remove them, but I'm not opposed if you want to remove them in some future PR.